### PR TITLE
[FLINK-27142][test] Remove bash tests dependencies on the Elasticsearch connector

### PR DESCRIPTION
**Pull request by alpreu**
## What is the purpose of the change

This is a draft PR  for the migration of the bash test relying on the Elasticsearch connector to java-based ones.

## Brief change log
- TODO

## Verifying this change

Please make sure both new and modified tests in this PR follows the conventions defined in our code quality guide: https://flink.apache.org/contributing/code-style-and-quality-common.html#testing

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (100MB)*
  - *Extended integration test for recovery after master (JobManager) failure*
  - *Added test that validates that TaskInfo is transferred only once across recoveries*
  - *Manually verified the change by running a 4 node cluster with 2 JobManagers and 4 TaskManagers, a stateful streaming program, and killing one JobManager and two TaskManagers during the execution, verifying that recovery happens correctly.*

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / no)
  - The serializers: (yes / no / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / no / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (yes / no / don't know)
  - The S3 file system connector: (yes / no / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by zentol** [comment]
Rebased and fixed all the issue I could find.

**Review by MartijnVisser — CHANGES_REQUESTED** [review]
The tests fail for me:

`java.lang.RuntimeException: java.io.FileNotFoundException: No resource file could be found that matches the pattern .*flink-test-utils.*\.jar. This could mean that the test module must be rebuilt via maven.`

**Comment by zentol** [comment]
> This could mean that the test module must be rebuilt via maven.

Did you follow this suggestion?

**Comment by MartijnVisser** [comment]
> This could mean that the test module must be rebuilt via maven.

Prior to running the test, I ran `mvn clean install -DskipTests -Dfast -Pskip-webui-build` on the entire project

File: flink-end-to-end-tests/flink-sql-client-test/src/test/java/SqlClientITCase.java [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+import java.nio.file.Path;
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Properties;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/** E2E Test for SqlClient. */
```
[the remark this answers is not in the document]
Comment (zentol) [reply flink-end-to-end-tests/flink-sql-client-test/src/test/java/SqlClientITCase.java:62]: I don't really see why we should rename it. It uses the SQL client to test certain use-cases, or iow test the sql-client end-to-end.
We'd have to rename 90% of our e2e tests to "smoke" tests if that wouldn't suffice.

**Comment by zentol** [comment]
What are the contents of `flink-end-to-end-tests/flink-sql-client-test/target/sql-jars`?

File: flink-end-to-end-tests/flink-sql-client-test/src/test/java/SqlClientITCase.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (MartijnVisser) [reply flink-end-to-end-tests/flink-sql-client-test/src/test/java/SqlClientITCase.java:62]: I think we're more testing the bundling of different things together to validate that this works over the actual working of the SQL Client. But you can consider it a nit.

**Comment by MartijnVisser** [comment]
Did a full `mvn clean` and a rebuild after, then the tests passed :/

**Review by MartijnVisser — APPROVED** [review]
LGTM

**Pull request merged**